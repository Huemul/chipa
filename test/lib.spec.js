const { parseFile, parsePath } = require('../lib')

describe('parseFile', () => {

  describe('when no second argument is passed', () => {
    it('extracts ALL the snippets from a file', () => {
      // glob uses process.cwd()
      // since jest is being run at the home of the project
      // the path has to be relative to that
      return parseFile('./test/snippets.md')
        .then(result => {
          expect(result).toMatchSnapshot()
        })
    })

  })

  describe('when a language is provided', () => {
    it('extracts the corresponding snippets from a file', () => {
      const ps = [
        parseFile('./test/snippets.md', ['js', 'javascript']),
        parseFile('./test/snippets.md', 'bash'),
        parseFile('./test/snippets.md', 'go'),
      ]
      return Promise.all(ps)
        .then(([js, bash, go]) => {
          expect(js).toMatchSnapshot()
          expect(bash).toMatchSnapshot()
          expect(go).toMatchSnapshot()
        })
    })
  })

  describe('when the file has no snippets', () => {
    it('resolves with empty snippets', () => {
      return parseFile('./test/no-code.md')
        .then(result => {
          expect(result.snippets.length).toBe(0)
        })
    })
  })

})

describe('parsePath', () => {

  describe('for a provided glob', () => {
    it('resolves with the snippets for each files & skips empty ones', () => {
      return parsePath('./test/*.md')
        .then(files => {
          expect(files).toMatchSnapshot()
        })
    })
  })

})

