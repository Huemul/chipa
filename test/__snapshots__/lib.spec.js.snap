// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extract for a provided glob resolves with the snippets for each files & skips empty ones 1`] = `
Array [
  Object {
    "file": "snippets-2.md",
    "lang": "all",
    "snippets": Array [
      Object {
        "lang": "js",
        "meta": "",
        "position": Object {
          "start": Object {
            "column": 1,
            "line": 3,
          },
        },
        "value": "console.log(\\"Hello, World!\\")",
      },
    ],
  },
  Object {
    "file": "snippets.md",
    "lang": "all",
    "snippets": Array [
      Object {
        "lang": "js",
        "meta": "(meta) (tags)",
        "position": Object {
          "start": Object {
            "column": 1,
            "line": 3,
          },
        },
        "value": "console.log(\\"Hello, World!\\")",
      },
      Object {
        "lang": "javascript",
        "meta": "",
        "position": Object {
          "start": Object {
            "column": 1,
            "line": 7,
          },
        },
        "value": "console.log(\\"Hello, World!\\")",
      },
      Object {
        "lang": "JavaScript",
        "meta": "",
        "position": Object {
          "start": Object {
            "column": 1,
            "line": 12,
          },
        },
        "value": "console.log(\\"Hello, World!\\")",
      },
      Object {
        "lang": "go",
        "meta": "",
        "position": Object {
          "start": Object {
            "column": 1,
            "line": 18,
          },
        },
        "value": "package main

import \\"fmt\\"

func main() {
  fmt.Println(\\"Hello, World!\\")
}",
      },
      Object {
        "lang": "golang",
        "meta": "",
        "position": Object {
          "start": Object {
            "column": 1,
            "line": 28,
          },
        },
        "value": "package main

import \\"fmt\\"

func main() {
  fmt.Println(\\"Hello, World!\\")
}",
      },
      Object {
        "lang": "bash",
        "meta": "",
        "position": Object {
          "start": Object {
            "column": 1,
            "line": 40,
          },
        },
        "value": "$ echo 'Hello, World!'",
      },
    ],
  },
]
`;

exports[`extractSingle when a language is provided extracts the corresponding snippets from a file 1`] = `
Object {
  "file": "snippets.md",
  "lang": Array [
    "js",
    "javascript",
  ],
  "snippets": Array [
    Object {
      "lang": "js",
      "meta": "(meta) (tags)",
      "position": Object {
        "start": Object {
          "column": 1,
          "line": 3,
        },
      },
      "value": "console.log(\\"Hello, World!\\")",
    },
    Object {
      "lang": "javascript",
      "meta": "",
      "position": Object {
        "start": Object {
          "column": 1,
          "line": 7,
        },
      },
      "value": "console.log(\\"Hello, World!\\")",
    },
  ],
}
`;

exports[`extractSingle when a language is provided extracts the corresponding snippets from a file 2`] = `
Object {
  "file": "snippets.md",
  "lang": "bash",
  "snippets": Array [
    Object {
      "lang": "bash",
      "meta": "",
      "position": Object {
        "start": Object {
          "column": 1,
          "line": 40,
        },
      },
      "value": "$ echo 'Hello, World!'",
    },
  ],
}
`;

exports[`extractSingle when a language is provided extracts the corresponding snippets from a file 3`] = `
Object {
  "file": "snippets.md",
  "lang": "go",
  "snippets": Array [
    Object {
      "lang": "go",
      "meta": "",
      "position": Object {
        "start": Object {
          "column": 1,
          "line": 18,
        },
      },
      "value": "package main

import \\"fmt\\"

func main() {
  fmt.Println(\\"Hello, World!\\")
}",
    },
    Object {
      "lang": "go",
      "meta": "lang",
      "position": Object {
        "start": Object {
          "column": 1,
          "line": 28,
        },
      },
      "value": "package main

import \\"fmt\\"

func main() {
  fmt.Println(\\"Hello, World!\\")
}",
    },
  ],
}
`;

exports[`extractSingle when no second argument is passed extracts ALL the snippets from a file 1`] = `
Object {
  "file": "snippets.md",
  "lang": "all",
  "snippets": Array [
    Object {
      "lang": "js",
      "meta": "(meta) (tags)",
      "position": Object {
        "start": Object {
          "column": 1,
          "line": 3,
        },
      },
      "value": "console.log(\\"Hello, World!\\")",
    },
    Object {
      "lang": "javascript",
      "meta": "",
      "position": Object {
        "start": Object {
          "column": 1,
          "line": 7,
        },
      },
      "value": "console.log(\\"Hello, World!\\")",
    },
    Object {
      "lang": "JavaScript",
      "meta": "",
      "position": Object {
        "start": Object {
          "column": 1,
          "line": 12,
        },
      },
      "value": "console.log(\\"Hello, World!\\")",
    },
    Object {
      "lang": "go",
      "meta": "",
      "position": Object {
        "start": Object {
          "column": 1,
          "line": 18,
        },
      },
      "value": "package main

import \\"fmt\\"

func main() {
  fmt.Println(\\"Hello, World!\\")
}",
    },
    Object {
      "lang": "golang",
      "meta": "",
      "position": Object {
        "start": Object {
          "column": 1,
          "line": 28,
        },
      },
      "value": "package main

import \\"fmt\\"

func main() {
  fmt.Println(\\"Hello, World!\\")
}",
    },
    Object {
      "lang": "bash",
      "meta": "",
      "position": Object {
        "start": Object {
          "column": 1,
          "line": 40,
        },
      },
      "value": "$ echo 'Hello, World!'",
    },
  ],
}
`;
